full-node/sov-ethereum/src/lib.rs

```diff
@@ -225,21 +225,74 @@ pub mod experimental {
                     .map(|id| id.as_u64())
                     .unwrap_or(1);
 
-                // TODO: implement gas logic after gas estimation (#906) is implemented
-                // https://github.com/Sovereign-Labs/sovereign-sdk/issues/906
+                // TODO: we need an oracle to fetch the gas price of the current chain
```

Comment:
`eth_sendTransaction` is quite long let' split it into several methods? We can also crate a file `eth_sendTransaction .rs` and move the code there.